pkg/kds/server/status_tracker.go

```diff
@@ -0,0 +1,191 @@
+package server
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/kds/util"
+
+	"github.com/go-logr/logr"
+
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+
+	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoy_xds "github.com/envoyproxy/go-control-plane/pkg/server/v2"
+	"github.com/golang/protobuf/proto"
+
+	"github.com/kumahq/kuma/pkg/core"
+	core_runtime "github.com/kumahq/kuma/pkg/core/runtime"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+var (
+	// overridable by unit tests
+	now     = time.Now
+	newUUID = core.NewUUID
```

Comment:
I know you were probably following the pattern, but we can just override `core.NewUUID` in tests and for time it should be `core.Now` which also can be overriden

Reply:
Fixed, thanks!